packages/@aws-cdk/cloudformation-include/test/test-templates/nested/parent-stack.expected.json

```diff
@@ -0,0 +1,178 @@
+{
+  "Resources": {
+    "ChildStack": {
+      "Type": "AWS::CloudFormation::Stack",
+      "Properties": {
+        "TemplateURL": "https://cfn-templates-set.s3.amazonaws.com/child-import-stack.json",
+        "Parameters": {
+          "MyBucketParameter": "some-magic-bucket-name"
+        }
+      }
+    },
+    "AnotherChildStack": {
+      "Type": "AWS::CloudFormation::Stack",
+      "Properties": {
+        "TemplateURL": "https://cfn-templates-set.s3.amazonaws.com/child-import-stack.json",
+        "Parameters": {
+          "MyBucketParameter": "some-magic-bucket-name"
+        }
+      }
+    },
+    "ParentStackNestedChildStackNestedStackNestedChildStackNestedStackResource191FA7EB": {
+      "Type": "AWS::CloudFormation::Stack",
+      "Properties": {
+        "TemplateURL": {
+          "Fn::Join": [
+            "",
+            [
+              "https://s3.",
+              {
+                "Ref": "AWS::Region"
+              },
+              ".",
+              {
+                "Ref": "AWS::URLSuffix"
+              },
+              "/",
+              {
+                "Ref": "AssetParameters08289cd73ff85427f741904b82f721b6186819766f3e6ed6ff53a9f663faa45fS3BucketC7E3CB0B"
+              },
+              "/",
+              {
+                "Fn::Select": [
+                  0,
+                  {
+                    "Fn::Split": [
+                      "||",
+                      {
+                        "Ref": "AssetParameters08289cd73ff85427f741904b82f721b6186819766f3e6ed6ff53a9f663faa45fS3VersionKey03B2C785"
+                      }
+                    ]
+                  }
+                ]
+              },
+              {
+                "Fn::Select": [
+                  1,
+                  {
+                    "Fn::Split": [
+                      "||",
+                      {
+                        "Ref": "AssetParameters08289cd73ff85427f741904b82f721b6186819766f3e6ed6ff53a9f663faa45fS3VersionKey03B2C785"
+                      }
+                    ]
+                  }
+                ]
+              }
+            ]
+          ]
+        },
+        "Parameters": {
+          "referencetoAssetParameters5dc7d4a99cfe2979687dc74f2db9fd75f253b5505a1912b5ceecf70c9aefba50S3BucketEAA24F0CRef": {
+            "Ref": "AssetParameters5dc7d4a99cfe2979687dc74f2db9fd75f253b5505a1912b5ceecf70c9aefba50S3BucketEAA24F0C"
+          },
+          "referencetoAssetParameters5dc7d4a99cfe2979687dc74f2db9fd75f253b5505a1912b5ceecf70c9aefba50S3VersionKey1194CAB2Ref": {
+            "Ref": "AssetParameters5dc7d4a99cfe2979687dc74f2db9fd75f253b5505a1912b5ceecf70c9aefba50S3VersionKey1194CAB2"
+          }
+        }
+      }
+    },
+    "ParentStackNestedAnotherChildStackNestedStackNestedAnotherChildStackNestedStackResourceB27371BC": {
```

Comment:
Uhm. I think there's some misunderstanding here.

Your functionality should change the `templateUrl` property of `ChildStack` and `AnotherChildStack`. Not create new nested stacks that are copies of them. That's not migrating anything.

The expected JSON here should be:

```json
{
  "Resources": {
    "ChildStack": {
      "Type": "AWS::CloudFormation::Stack",
      "Properties": {
        "TemplateURL": {
          "Fn::Join": [
            "",
            [
              "https://s3.",
              {
                "Ref": "AWS::Region"
              },
              ".",
              {
                "Ref": "AWS::URLSuffix"
              },
              "/",
              {
                "Ref": "AssetParameters08289cd73ff85427f741904b82f721b6186819766f3e6ed6ff53a9f663faa45fS3BucketC7E3CB0B"
              },
              "/",
              {
                "Fn::Select": [
                  0,
                  {
                    "Fn::Split": [
                      "||",
                      {
                        "Ref": "AssetParameters08289cd73ff85427f741904b82f721b6186819766f3e6ed6ff53a9f663faa45fS3VersionKey03B2C785"
                      }
                    ]
                  }
                ]
              },
              {
                "Fn::Select": [
                  1,
                  {
                    "Fn::Split": [
                      "||",
                      {
                        "Ref": "AssetParameters08289cd73ff85427f741904b82f721b6186819766f3e6ed6ff53a9f663faa45fS3VersionKey03B2C785"
                      }
                    ]
                  }
                ]
              }
            ]
          ]
        },
        "Parameters": {
          "MyBucketParameter": "some-magic-bucket-name",
          "referencetoAssetParameters5dc7d4a99cfe2979687dc74f2db9fd75f253b5505a1912b5ceecf70c9aefba50S3BucketEAA24F0CRef": {
            "Ref": "AssetParameters5dc7d4a99cfe2979687dc74f2db9fd75f253b5505a1912b5ceecf70c9aefba50S3BucketEAA24F0C"
          },
          "referencetoAssetParameters5dc7d4a99cfe2979687dc74f2db9fd75f253b5505a1912b5ceecf70c9aefba50S3VersionKey1194CAB2Ref": {
            "Ref": "AssetParameters5dc7d4a99cfe2979687dc74f2db9fd75f253b5505a1912b5ceecf70c9aefba50S3VersionKey1194CAB2"
          }
        }
      }
    },
    "AnotherChildStack": {
      "Type": "AWS::CloudFormation::Stack",
      "Properties": {
        "TemplateURL": {
          "Fn::Join": [
            "",
            [
              "https://s3.",
              {
                "Ref": "AWS::Region"
              },
              ".",
              {
                "Ref": "AWS::URLSuffix"
              },
              "/",
              {
                "Ref": "AssetParameters10881a48ed8be2f8ba535e786796d686eddc3a4df9f76ef9bb549b1dcbd1f69eS3Bucket95AE9A81"
              },
              "/",
              {
                "Fn::Select": [
                  0,
                  {
                    "Fn::Split": [
                      "||",
                      {
                        "Ref": "AssetParameters10881a48ed8be2f8ba535e786796d686eddc3a4df9f76ef9bb549b1dcbd1f69eS3VersionKey01024429"
                      }
                    ]
                  }
                ]
              },
              {
                "Fn::Select": [
                  1,
                  {
                    "Fn::Split": [
                      "||",
                      {
                        "Ref": "AssetParameters10881a48ed8be2f8ba535e786796d686eddc3a4df9f76ef9bb549b1dcbd1f69eS3VersionKey01024429"
                      }
                    ]
                  }
                ]
              }
            ]
          ]
        },
        "Parameters": {
          "Parameters": {
            "MyBucketParameter": "some-magic-bucket-name"
          },
          "referencetoAssetParameters5dc7d4a99cfe2979687dc74f2db9fd75f253b5505a1912b5ceecf70c9aefba50S3BucketEAA24F0CRef": {
            "Ref": "AssetParameters5dc7d4a99cfe2979687dc74f2db9fd75f253b5505a1912b5ceecf70c9aefba50S3BucketEAA24F0C"
          },
          "referencetoAssetParameters5dc7d4a99cfe2979687dc74f2db9fd75f253b5505a1912b5ceecf70c9aefba50S3VersionKey1194CAB2Ref": {
            "Ref": "AssetParameters5dc7d4a99cfe2979687dc74f2db9fd75f253b5505a1912b5ceecf70c9aefba50S3VersionKey1194CAB2"
          }
        }
      }
    }
  },
  "Parameters": {
    "AssetParameters5dc7d4a99cfe2979687dc74f2db9fd75f253b5505a1912b5ceecf70c9aefba50S3BucketEAA24F0C": {
      "Type": "String",
      "Description": "S3 bucket for asset \"5dc7d4a99cfe2979687dc74f2db9fd75f253b5505a1912b5ceecf70c9aefba50\""
    },
    "AssetParameters5dc7d4a99cfe2979687dc74f2db9fd75f253b5505a1912b5ceecf70c9aefba50S3VersionKey1194CAB2": {
      "Type": "String",
      "Description": "S3 key for asset version \"5dc7d4a99cfe2979687dc74f2db9fd75f253b5505a1912b5ceecf70c9aefba50\""
    },
    "AssetParameters5dc7d4a99cfe2979687dc74f2db9fd75f253b5505a1912b5ceecf70c9aefba50ArtifactHash9C417847": {
      "Type": "String",
      "Description": "Artifact hash for asset \"5dc7d4a99cfe2979687dc74f2db9fd75f253b5505a1912b5ceecf70c9aefba50\""
    },
    "AssetParameters08289cd73ff85427f741904b82f721b6186819766f3e6ed6ff53a9f663faa45fS3BucketC7E3CB0B": {
      "Type": "String",
      "Description": "S3 bucket for asset \"08289cd73ff85427f741904b82f721b6186819766f3e6ed6ff53a9f663faa45f\""
    },
    "AssetParameters08289cd73ff85427f741904b82f721b6186819766f3e6ed6ff53a9f663faa45fS3VersionKey03B2C785": {
      "Type": "String",
      "Description": "S3 key for asset version \"08289cd73ff85427f741904b82f721b6186819766f3e6ed6ff53a9f663faa45f\""
    },
    "AssetParameters08289cd73ff85427f741904b82f721b6186819766f3e6ed6ff53a9f663faa45fArtifactHashC9440392": {
      "Type": "String",
      "Description": "Artifact hash for asset \"08289cd73ff85427f741904b82f721b6186819766f3e6ed6ff53a9f663faa45f\""
    },
    "AssetParameters10881a48ed8be2f8ba535e786796d686eddc3a4df9f76ef9bb549b1dcbd1f69eS3Bucket95AE9A81": {
      "Type": "String",
      "Description": "S3 bucket for asset \"10881a48ed8be2f8ba535e786796d686eddc3a4df9f76ef9bb549b1dcbd1f69e\""
    },
    "AssetParameters10881a48ed8be2f8ba535e786796d686eddc3a4df9f76ef9bb549b1dcbd1f69eS3VersionKey01024429": {
      "Type": "String",
      "Description": "S3 key for asset version \"10881a48ed8be2f8ba535e786796d686eddc3a4df9f76ef9bb549b1dcbd1f69e\""
    },
    "AssetParameters10881a48ed8be2f8ba535e786796d686eddc3a4df9f76ef9bb549b1dcbd1f69eArtifactHash7A7DB969": {
      "Type": "String",
      "Description": "Artifact hash for asset \"10881a48ed8be2f8ba535e786796d686eddc3a4df9f76ef9bb549b1dcbd1f69e\""
    }
  }
}
```